cmd/controller-manager/app/serve.go

```diff
@@ -53,9 +52,9 @@ func BuildHandlerChain(apiHandler http.Handler, authorizationInfo *apiserver.Aut
 }
 
 // NewBaseHandler takes in CompletedConfig and returns a handler.
-func NewBaseHandler(c *apiserverconfig.DebuggingConfiguration) *mux.PathRecorderMux {
+func NewBaseHandler(c *apiserverconfig.DebuggingConfiguration, checks ...healthz.HealthzChecker) *mux.PathRecorderMux {
```

Comment:
Is it easy to add checks to the DebuggingConfiguration (which sounds misnamed)? That's more backwards compatible.

Don't bother if it doesn't actually make sense or isn't easy.

Reply:
Not so much hard as I would like to limit the scope of the change to the controller managers. This would effect other things like scheduler and I'm not sure scheduler needs the same sort of wiring.